src/Storages/StorageBuffer.cpp

```diff
@@ -842,9 +843,22 @@ void registerStorageBuffer(StorageFactory & factory)
                 " destination_database, destination_table, num_buckets, min_time, max_time, min_rows, max_rows, min_bytes, max_bytes.",
                 ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
+        // Table and database name arguments accept expressions, evaluate them.
         engine_args[0] = evaluateConstantExpressionForDatabaseName(engine_args[0], args.local_context);
         engine_args[1] = evaluateConstantExpressionOrIdentifierAsLiteral(engine_args[1], args.local_context);
 
+        // After we evaluated all expressions, check that all arguments are
+        // literals.
+        for (size_t i = 0; i < 9; i++)
```

Comment:
```suggestion
        for (size_t i = 2; i < 9; i++)
```
If it doesn't become a literal after evaluation, `as<ASTLiteral &>()` will throw `LOGICAL_ERROR`

Reply:
In this case, `typeid_cast` in the loop should return nullptr, and we'll throw a BAD_ARGUMENTS exception. Looks OK. Btw, do you have any example for this? I couldn't construct one.